magda-web-client/src/Components/Dataset/Add/DatasetAddMetadataPage.tsx

```diff
@@ -333,11 +356,22 @@ class NewDataset extends React.Component<Props, State> {
 
             await this.resetError();
 
-            this.setState(state => ({
+            this.setState((state) => ({
                 ...state,
                 isPublishing: true
             }));
 
+            // Setting datasets as approved if
+            // approval flow is turned off
+            if (!config.featureFlags.datasetApprovalWorkflowOn) {
```

Comment:
@AlexGilleran Currently, when `datasetApprovalWorkflowOn` is off, the submitted dataset is still saved with `draft` status. Because of, this dataset will not be shown under either `draft` or `published` dataset tab (as it has no `dataset-draft` aspect but status != published). People would think the dataset is disappeared. Do you mind if I change this to always set the dataset to `published`?
Alternatively, we can change the `draft` dataset tab query and make `dataset-draft` aspect and optionalAspect?
The way of updating publishing status may also have a timing issue:
The `this.setState` may not be completed immediately before  `await submitDatasetFromState`. I probably rewrite this part as well.

Reply:
Sure!